MapleServer2/Types/Currency.cs

```diff
@@ -0,0 +1,70 @@
+﻿using MapleServer2.Enums;
+using MapleServer2.Packets;
+
+namespace MapleServer2.Types
+{
+    public class Currency
+    {
+        private readonly Player Player;
+        private readonly CurrencyType Type;
+        public long Amount { get; private set; }
+
+        public Currency(Player player, CurrencyType type, long input)
+        {
+            Player = player;
+            Type = type;
+            Amount = input;
+        }
+
+        public bool Modify(long input)
```

Comment:
Seems like overflow is not handled here.

Reply:
if (Amount + input < 0) is the overflow handling

Reply:
Ah right I forgot that overflow just turns negative.